src/main/java/dev/morling/onebrc/CompareWithTolerance.java

```diff
@@ -0,0 +1,105 @@
+/*
+ *  Copyright 2023 The original authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package dev.morling.onebrc;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.regex.Pattern;
+import java.util.stream.IntStream;
+
+public class CompareWithTolerance {
+    private static final Pattern numRegex = Pattern.compile("-?\\d\\d?\\.\\d");
+
+    record NameAndStats(String name, int[] stats) {
+    }
+
+    public static void main(String[] args) throws IOException {
+        var fname1 = args[0];
+        var fname2 = args[1];
+        var entries1 = entries(fname1);
+        var entries2 = entries(fname2);
+        if (entries1.length != entries2.length) {
+            System.out.printf("Files don't match in the number of entries: %,d vs. %,d%n", entries1.length, entries2.length);
+            System.exit(1);
+        }
+        for (int i = 0; i < entries1.length; i++) {
+            var nameAndStats1 = parseEntry(entries1[i], i, fname1);
+            var nameAndStats2 = parseEntry(entries2[i], i, fname2);
+            if (IntStream.range(0, 3)
+                    .mapToObj(n -> new double[]{ nameAndStats1.stats[n], nameAndStats2.stats[n] })
+                    .anyMatch(pair -> Math.abs((pair[0] - pair[1])) > 1)) {
```

Comment:
I think the baseline would only ever get the value too high by 0.1 (or 1 as `int` here), i.e. no need for `abs()`?